core/time/new_spec.rb

```diff
@@ -475,4 +475,94 @@ def zone.local_to_utc(t)
       end
     end
   end
+
+  ruby_version_is "3.2" do
+    it "parses a time string" do
+      t = Time.utc(2020, 12, 24, 15, 56, 17)
+
+      Time.new("2020-12-24T15:56:17Z").should == t
+      Time.new("2020-12-25 00:56:17 +09:00").should == t
+      Time.new("2020-12-25 00:57:47 +09:01:30").should == t
+      Time.new("2020-12-25 00:56:17 +0900").should == t
+      Time.new("2020-12-25 00:57:47 +090130").should == t
+      Time.new("2020-12-25T00:56:17+09:00").should == t
+    end
+
+    it "accepts precission keyword argument" do
+      Time.new("2021-12-25 00:00:00.123456789876 +09:00", precision: 3).subsec.should == 0.123r
+      Time.new("2021-12-25 00:00:00.123456789876 +09:00", precision: nil).subsec.should == 0.123456789876r
+    end
+
+    it "raises ArgumentError if part of time string is missing" do
+      -> {
+        Time.new("2020-12-25 00:56 +09:00")
+      }.should raise_error(ArgumentError, /missing sec part/)
+
+      -> {
+        Time.new("2020-12-25 00 +09:00")
+      }.should raise_error(ArgumentError, /missing min part/)
+    end
+
+    it "raises ArgumentError if zone argument is passed without in: keyword" do
+      -> {
+        Time.new(2021, 1, 1, "+09:99")
+      }.should raise_error(ArgumentError, /invalid value for Integer()/)
+    end
```

Comment:
This test looks like not related to String argument parsing. If you would like to add it - it's better to add in a separate commit